src/Http/OpenApi/src/OpenApiGenerator.cs

```diff
@@ -0,0 +1,472 @@
+            var nullability = nullabilityContext.Create(parameter);
+            var isOptional = parameter.HasDefaultValue || nullability.ReadState != NullabilityState.NotNull;
+            var openApiParameter = new OpenApiParameter()
+            {
+                Name = parameter.Name,
+                In = parameterLocation,
+                Content = GetOpenApiParameterContent(metadata),
+                Schema = new OpenApiSchema { Type = SchemaGenerator.GetOpenApiSchemaType(parameter.ParameterType) },
+                Required = !isOptional
+
+            };
+            openApiParameters.Add(openApiParameter);
+        }
+
+        return openApiParameters;
+    }
+
+    private static IDictionary<string, OpenApiMediaType> GetOpenApiParameterContent(EndpointMetadataCollection metadata)
+    {
+        var openApiParameterContent = new Dictionary<string, OpenApiMediaType>();
+        var acceptsMetadata = metadata.GetMetadata<IAcceptsMetadata>();
+        if (acceptsMetadata is not null)
+        {
+            foreach (var contentType in acceptsMetadata.ContentTypes)
+            {
+                openApiParameterContent.Add(contentType, new OpenApiMediaType());
+            }
+        }
+
+        return openApiParameterContent;
+    }
+
+    private (bool, ParameterLocation?) GetOpenApiParameterLocation(ParameterInfo parameter, RoutePattern pattern, bool disableInferredBody)
+    {
+        var attributes = parameter.GetCustomAttributes();
+
+        if (attributes.OfType<IFromRouteMetadata>().FirstOrDefault() is { } routeAttribute)
+        {
+            return (false, ParameterLocation.Path);
+        }
+        else if (attributes.OfType<IFromQueryMetadata>().FirstOrDefault() is { } queryAttribute)
+        {
+            return (false, ParameterLocation.Query);
+        }
+        else if (attributes.OfType<IFromHeaderMetadata>().FirstOrDefault() is { } headerAttribute)
+        {
+            return (false, ParameterLocation.Header);
+        }
+        else if (attributes.OfType<IFromBodyMetadata>().FirstOrDefault() is { } fromBodyAttribute)
+        {
+            return (true, null);
+        }
+        else if (attributes.OfType<IFromFormMetadata>().FirstOrDefault() is { } fromFormAttribute)
+        {
+            return (true, null);
+        }
+        else if (parameter.CustomAttributes.Any(a => typeof(IFromServiceMetadata).IsAssignableFrom(a.AttributeType)) ||
+                parameter.ParameterType == typeof(HttpContext) ||
+                parameter.ParameterType == typeof(HttpRequest) ||
+                parameter.ParameterType == typeof(HttpResponse) ||
+                parameter.ParameterType == typeof(ClaimsPrincipal) ||
+                parameter.ParameterType == typeof(CancellationToken) ||
+                ParameterBindingMethodCache.HasBindAsyncMethod(parameter) ||
+                _serviceProviderIsService?.IsService(parameter.ParameterType) == true)
```

Comment:
nit
```suggestion
                _serviceProviderIsService?.IsService(parameter.ParameterType))
```

Reply:
`?.IsService` will return a `bool?` so we need to do an explicit check to get a `bool` here.